### PR TITLE
Task/DES-1552 - Download Published File Archives

### PR DESCRIPTION
**Pull request by Strmiska**
Created a "Publication Download" modal and updated a few of the templates. This should download any archive created for a publication or at least show an error message for the user to submit a ticket.

This can be tested locally if you have some published projects available in your index. Just make sure that the publication has an existing archive saved on corral. The path should look something like:
`/corral-repl/projects/NHERI/published/archives/PRJ-XXXX_archive.zip`

### CODE REVIEW

**Comment by rstijerina** [comment]
Is there a good example of a project that has this button? What would the url be for local dev?

**Comment by Strmiska** [comment]
> @Strmiska Is there a good example of a project that has this button? What would the url be for local dev?

Sorry I should have posted that along with an image of where this button is! If you go into any of the published projects you will see a "Download Dataset" button at the top right next to the title. If you open the modal inside a published project it should show an error if the archive is missing (this should happen for a few publications locally). I think the first few experiments in the published listing should give a download option.

**Comment by rstijerina** [comment]
>  I think the first few experiments in the published listing should give a download option.

Can you link me? Not managing to find one, but maybe it's my local setup?

**Comment by Strmiska** [comment]
> > I think the first few experiments in the published listing should give a download option.
> 
> Can you link me? Not managing to find one, but maybe it's my local setup?

This should work, but if you can't preview the published files locally then you may not be able to see the archives as well.
https://designsafe.dev/data/browser/public/designsafe.storage.published//PRJ-2537

**Comment by rstijerina** [comment]
> This should work, but if you can't preview the published files locally then you may not be able to see the archives as well.
> https://designsafe.dev/data/browser/public/designsafe.storage.published//PRJ-2537

Got it, thanks! Had to rebuild

File: designsafe/static/scripts/projects/components/publication-download/publication-download.component.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            this.arcData = resp;
+        }).finally(() => {
+            if (!this.arcData) {
+                this.error = true;
+            }
+            this.loaded = true;
+        });
+    }
+
+    download() {
```
Comment (rstijerina) [inline designsafe/static/scripts/projects/components/publication-download/publication-download.component.js]: can you add a spinner to the download button and disable the button while this function is running?

something like on `download`
`this.initiatingDownload = true;`

and in the template

```<button class="btn btn-add" ng-click="$ctrl.download()" ng-disabled="!$ctrl.loaded">
            <div class="btn-container">
                <i class="fa fa-spinner fa-spin" />
                <span class="curation-download"></span>
                &ensp;
                <span><strong>Download</strong></span>
            </div>
        </button>```

Comment (Strmiska) [reply designsafe/static/scripts/projects/components/publication-download/publication-download.component.js:35]: Sure, do you mean to show it spinning while the file is downloading or just while retrieving the agave postit? I think I we close the modal when the file starts to download don't you think. (these details weren't provided in the mockup)

Comment (rstijerina) [reply designsafe/static/scripts/projects/components/publication-download/publication-download.component.js:35]: Just while retrieving the postit. I think the rest of the behavior makes sense - closing the modal after the download starts

**Pull request merged**